torch/expanded_weights/layer_norm_expanded_weights.py

```diff
@@ -0,0 +1,37 @@
+
+import torch
+import torch.nn.functional as F
+from torch.expanded_weights.expanded_weights_impl import implements_per_sample_grads
+from torch.expanded_weights.expanded_weights_utils import\
+    forward_helper, grad_if_exists, grad_if_exists_for_input, sum_over_all_but_batch_and_last_n
+
+@implements_per_sample_grads(F.layer_norm)
+class LayerNormFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, *expanded_args):
+        input = expanded_args[0]
+        normalized_shape = expanded_args[1]
+        assert len(input.shape) > len(normalized_shape)
```

Comment:
A lot of these assertions (in other functions as well) should probably be RuntimeErrors so that they look nicer to the user